sdk/webpubsub/azure-messaging-webpubsubservice/azure/messaging/webpubsubservice/_configuration.py

```diff
@@ -0,0 +1,71 @@
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# Code generated by Microsoft (R) AutoRest Code Generator.
+# Changes may cause incorrect behavior and will be lost if the code is regenerated.
+# --------------------------------------------------------------------------
+
+from typing import TYPE_CHECKING
+
+from azure.core.configuration import Configuration
+from azure.core.pipeline import policies
+
+from ._version import VERSION
+from ._policies import JwtCredentialPolicy, ApiManagementProxy
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any, Optional, Union
+
+    from azure.core.credentials import TokenCredential, AzureKeyCredential
+
+
+def _proxy_policy(**kwargs):
+    if kwargs.get('proxy_endpoint'):
+        return ApiManagementProxy(kwargs.get('endpoint'), kwargs.get('proxy_endpoint'))
+    return None
+
+
+class WebPubSubServiceClientConfiguration(Configuration):
+    """Configuration for WebPubSubServiceClient.
+
+    Note that all parameters used to create this instance are saved as instance
+    attributes.
+
+    :param credential: Credential needed for the client to connect to Azure.
+    :type credential: Union[~azure.core.credentials.TokenCredential, ~azure.core.credentials.AzureKeyCredential]
+    """
+
+    def __init__(
+        self,
+        credential,  # type: Union[TokenCredential, AzureKeyCredential]
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        super(WebPubSubServiceClientConfiguration, self).__init__(**kwargs)
+
+        self.credential = credential
+        self.credential_scopes = kwargs.pop('credential_scopes', ['https://webpubsub.azure.com/.default'])
+        kwargs.setdefault('sdk_moniker', 'messaging-webpubsubservice/{}'.format(VERSION))
+        self._configure(**kwargs)
+
+    def _configure(
+        self,
+        **kwargs  # type: Any
+    ):
+        # type: (...) -> None
+        self.user_agent_policy = kwargs.get('user_agent_policy') or policies.UserAgentPolicy(**kwargs)
+        self.headers_policy = kwargs.get('headers_policy') or policies.HeadersPolicy(**kwargs)
+        self.proxy_policy = _proxy_policy(**kwargs) or policies.ProxyPolicy(**kwargs)
+        self.logging_policy = kwargs.get('logging_policy') or policies.NetworkTraceLoggingPolicy(**kwargs)
+        self.http_logging_policy = kwargs.get('http_logging_policy') or policies.HttpLoggingPolicy(**kwargs)
+        self.retry_policy = kwargs.get('retry_policy') or policies.RetryPolicy(**kwargs)
+        self.custom_hook_policy = kwargs.get('custom_hook_policy') or policies.CustomHookPolicy(**kwargs)
+        self.redirect_policy = kwargs.get('redirect_policy') or policies.RedirectPolicy(**kwargs)
+        self.authentication_policy = kwargs.get('authentication_policy')
+        if self.credential and not self.authentication_policy:
+            if kwargs.get("accesskey"):
```

Comment:
We should switch auth policy based on the credential type and not on the presence of the `accesskey` kwarg. We should not expose an `accesskey` in our client constructor.